syntax = "proto3";
package Sam;

message mutation {
    int32 position = 2;
    int32 ref_nuc = 3;
    int32 par_nuc = 4;
    int32 mut_nuc = 5;
    bool is_missing = 6;
}

message read_info {
    string read = 1;
    repeated mutation mutations = 2;
    int32 start_idx = 3;
    int32 end_idx = 4;
    int32 degree = 5; // how many repeats
}

message column_info {
    string column_name = 1;
    repeated string input_columns = 2;
}

message sam {
    repeated read_info reads = 1;
    repeated column_info reverse_columns = 2;
}